CMSgov/onemac
# OY2 9775 Regression Testing - Submission completed message is not displayed for phone number update

Story: https://qmacbis.atlassian.net/browse/OY2-9775
Endpoint: https://dzs1kup0nyiew.cloudfront.net/

### Details
Changes made to the AlertBar did not get propagated into all places where the AlertBar is being used. Also, the Success message on the phone number editing was not originally part of the story, so the decision was to remove it entirely.

While working on the Alert Bar, I took the opportunity to clean up our RESPONSE_CODE implementation.

### Changes
- Removed response-code.js file from app-api and set the scripts to include the cmscommonlib instead.
- Cleaned up RESPONSE_CODE list -- removed codes not being used, verified codes being used map to appropriate front end alerts, corrected some misspellings/miswordings
- Renamed alert-messages.js to alertsLib to following naming convention and allow conceptual consolidation of alert functionality.
- Converted getAlert function in front end to alertCodeAlerts lookup in alertsLib and removed error-mappings.js -- still need the mapping because there is a one-to-many relationship where several RESPONSE_CODEs map to the same alert.
- Used RESPONSE_CODE.NONE instead of hard-coding “NONE” for no alert.
- Updated Unit Tests to use the new Alert lookup structure.
- AlertBar closing:
  - SubmissionForm: Done before this story
  - Dashboard: added closeCallback to clear alertCode properly
  - Home: No Change, this page does not maintain its own errors, merely shows what comes through in the calling state, so unless we change that the close button creates a new historical event, nothing to do here :) 
  - UserManagement: added closeCallback to clear alertCode properly
  - UserPage: added closeCallback and also change to NONE if success (per story)
  - DevLogin: updated to use AlertBar
- Other fixes included:
- Submit.js -- Moved throw error in the spa-ids so that it only throws the error if a non-expected error occurs (we expect that some of the base ids are found and want to make sure the loop finishes adding them all)
- Duplicate RAI submissions - changed the order of limitSubmit and setIsSubmitting to prevent the situation where isSubmitting is true and limitSubmit is again false.  I would love a more stable solution for this!!!
- Cleaned up React warnings:
  - Removed lodash import from ChoiceList
  - Removed value dependency from PhoneNumber.onEditFn
  - Removed plusSign and negativeSign import from TriggerCB
  - Removed extraneous variable ‘_’ from Signup
  - Removed getAdminTypeByRole import from UserManagement
  - Removed helpDeskContact import from UserPage
  - Added dependencies for userData.group and userData.type in UserPage
  - Removed formatDate import from tableListExportToCSV

- HA!  And FINALLY - set alertCode to RESPONSE_CODE.NONE when success is returned from a phone number update!!!!
- Added a check to UserPage Unit Test to verify that success alert is not showing after successful submit.

### Implementation Notes
YES, a lot of files have been changed with VERY SMALL CHANGES :) 

### Test Plan
1. Login as any user
2. Travel to UserPage
3. If phone number exists, click Edit, otherwise, click Add
4. Enter information into field and click Submit
5. No Success message should be displayed
6. Turn off Wifi and click Submit to get error message.
7. Close Alert and re-submit to see error message again.
